pkg/subctl/cmd/verify.go

```diff
@@ -0,0 +1,106 @@
+/*
+© 2020 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/onsi/ginkgo/config"
+	"github.com/spf13/cobra"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+
+	verifyconnectivity "github.com/submariner-io/submariner-operator/pkg/subctl/verify/connectivity"
+	verifyservicediscovery "github.com/submariner-io/submariner-operator/pkg/subctl/verify/servicediscovery"
+)
+
+var (
+	verifyConnectivity              bool
+	verifyServiceDiscovery          bool
+	verifyAll                       bool
+	verboseConnectivityVerification bool
+	operationTimeout                uint
+	connectionTimeout               uint
+	connectionAttempts              uint
+	reportDirectory                 string
+)
+
+func init() {
+	addVerifyFlags(verifyCmd)
+	rootCmd.AddCommand(verifyCmd)
+}
+
+func addVerifyFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVar(&verifyConnectivity, "connectivity", true, "verify connectivity between two clusters")
+	cmd.Flags().BoolVar(&verifyServiceDiscovery, "service-discovery", false, "verify service-discovery between two clusters")
+	cmd.Flags().BoolVar(&verifyAll, "all", false, "verify connectivity and service-discovery between two clusters")
```

Comment:
Those three are particular to this command only, not the old deprecated one, can we move those 3 to the init?

Reply:
I thought I did that, likely forgot to commit before pushing and cleaned it up. will do.